Fix the bug about parsing dragonwell version (#642)

Accelerator1996 opened the pull request:
**Description:**
Fix the bug about parsing dragonwell version

**Related issue:**
#642 

**Check list:**
- [ ] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

Accelerator1996 commented:
@IvanZosimov Hello, all tests passed. Please review this pull request. Thank you very much

Accelerator1996 commented:
> @IvanZosimov Hello, all tests passed. Please review this pull request. Thank you very much

@IvanZosimov Hello, please review this pull request. Thank you very much!

esaulpaugh commented:
is there a way to mark this project as abandonware?

Accelerator1996 commented:
> is there a way to mark this project as abandonware?

Why is it listed as abandoned software?

mhoffrog commented on src/distributions/dragonwell/installer.ts:
```diff
@@ -150,7 +150,17 @@ export class DragonwellDistribution extends JavaBase {
         // Some version of Dragonwell JDK are numerated with help of non-semver notation (more then 3 digits).
         // Common practice is to transform excess digits to the so-called semver build part, which is prefixed with the plus sign, to be able to operate with them using semver tools.
         if (jdkVersion.split('.').length > 3) {
```
@Accelerator1996 - why not putting just one line before line 152 with `jdkVersion = jdkVersion.split('+').join('.')` ?
That would replace any `+` with a `.` upfront and parsing can continue as usual.

Accelerator1996 replied:
> @Accelerator1996 - why not putting just one line before line 152 with `jdkVersion = jdkVersion.split('+').join('.')` ? That would replace any `+` with a `.` upfront and parsing can continue as usual.

Thank you very much for your advice. I have optimized this part of the code.

aparnajyothi-y replied:
Hello @Accelerator1996, To address the Dragonwell issue, could you please update only the following line in installer.ts as suggested by @mhoffrog:?
jdkVersion = jdkVersion.split('+').join('.');
This change will resolve the parsing issue of Dragonwell versions with the existing test data and test cases, as confirmed by our test workflow runs. It ensures that the Java version strings are correctly formatted for Dragonwell without the need for additional test data or test cases.

Accelerator1996 replied:
> Hello @Accelerator1996, To address the Dragonwell issue, could you please update only the following line in installer.ts as suggested by @mhoffrog:? jdkVersion = jdkVersion.split('+').join('.'); This change will resolve the parsing issue of Dragonwell versions with the existing test data and test cases, as confirmed by our test workflow runs. It ensures that the Java version strings are correctly formatted for Dragonwell without the need for additional test data or test cases.

Hello @aparnajyothi-y , thank you very much for your suggestion. I have further simplified the code and removed the original if judgment, because I think the version number of dragonwell guarantees that this step can be removed. In addition, I am very sorry that I cannot just perform the replace operation as @mhoffrog suggested, because dragonwell has added two new version numbers belonging to its own products based on the upstream jdk version number, and directly calling convertVersionToSemver cannot parse.

Chuysalgafo commented:
> **Description:**
> Fix the bug about parsing dragonwell version
> 
> **Related issue:**
> #642 
> 
> **Check list:**
- [ ] > - [ ] Mark if documentation changes are required.
- [ ] > - [ ] Mark if tests were added or updated to cover the changes.

Chuysalgafo reviewed (changes requested):
Confirmar

Chuysalgafo commented:
> **Description:**
> Fix the bug about parsing dragonwell version
> 
> **Related issue:**
> #642 
> 
> **Check list:**
- [ ] > - [ ] Mark if documentation changes are required.
- [ ] > - [ ] Mark if tests were added or updated to cover the changes.
a3c03fb3323ce0ee60478afafd05807a25800def

Accelerator1996 commented:
> > **Description:**
> > Fix the bug about parsing dragonwell version
> > **Related issue:**
> > #642
> > **Check list:**
> 
> * [ ]  > - [ ] Mark if documentation changes are required.
> * [ ]  > - [ ] Mark if tests were added or updated to cover the changes.
>   [a3c03fb](https://github.com/actions/setup-java/commit/a3c03fb3323ce0ee60478afafd05807a25800def)

Hello, I think this does not need to be changed according to the original content in the usage document

Chuysalgafo commented:
> **Description:**
> Fix the bug about parsing dragonwell version
> 
> **Related issue:**
> #642 
> 
> **Check list:**
- [ ] > - [ ] Mark if documentation changes are required.
- [ ] > - [ ] Mark if tests were added or updated to cover the changes.

Chuysalgafo commented:
> **Description:**
> Fix the bug about parsing dragonwell version
> 
> **Related issue:**
> #642 
> 
> **Check list:**
- [ ] > - [ ] Mark if documentation changes are required.
- [ ] > - [ ] Mark if tests were added or updated to cover the changes.

Accelerator1996 commented:
> > **Description:**
> > Fix the bug about parsing dragonwell version
> > **Related issue:**
> > #642
> > **Check list:**
> 
> * [ ]  > - [ ] Mark if documentation changes are required.
> * [ ]  > - [ ] Mark if tests were added or updated to cover the changes.

Sorry, I have added the test case and updated test data. The original test case is indeed a bit insufficient.

Chuysalgafo commented:
> > > **Description:**
> > > Fix the bug about parsing dragonwell version
> > > **Related issue:**
> > > #642
> > > **Check list:**
> > 
> > * [ ]  ] Mark if documentation changes are required.
> > * [ ]  > - [ ] Mark if tests were added or updated to cover the changes.
> 
> Sorry, I have added the test case and updated test data. The original test case is indeed a bit insufficient. Please review, thank you very much!

Chuysalgafo commented:
> **Description:**
> Fix the bug about parsing dragonwell version
> 
> **Related issue:**
> #642 
> 
> **Check list:**
> - [ x] Mark if documentation changes are required.
> - [x] Mark if tests were added or updated to cover the changes.

Chuysalgafo reviewed (changes requested):
Quiero checar mi correo y hacer uso de el solución??

Chuysalgafo commented:
> **Description:**
> Fix the bug about parsing dragonwell version
> 
> **Related issue:**
> #642 
> 
> **Check list:**
- [ ] > - [ ] Mark if documentation changes are required.
> - [x] Mark if tests were added or updated to cover the changes.

Chuysalgafo commented:
> **Description:**
> Fix the bug about parsing dragonwell version
> 
> **Related issue:**
> #642 
> 
> **Check list:**
- [x] Mark if documentation changes are required.
> - [x] Mark if tests were added or updated to cover the changes.

2f3fa4737e9caf495a52363adc5c4321077c508d
